### PR TITLE
Use "denotes a type" in utilities library clause

### PR DESCRIPTION
**Pull request by jwakely**
This replaces #2354, in line with the editorial meeting decision.

### CODE REVIEW

File: source/utilities.tex [hunk 1]
```diff
@@ -17527,7 +17527,7 @@
 \indexlibrary{\idxcode{rank}}%
 \tcode{template<class T>\br
  struct rank;}      &
- If \tcode{T} names an array type, an integer value representing
+ If \tcode{T} denotes an array type, an integer value representing
```
Comment (zygoloid) [inline source/utilities.tex]: I think for template parameters we just want "is"; "denotes" should be used for typedefs.

Comment (jwakely) [reply source/utilities.tex]: Done (here and elsewhere).

**Comment by jwakely** [comment]
I've rebased it on `master`, and made the requested changes.

File: source/utilities.tex [hunk 2]
```diff
@@ -17992,18 +17993,18 @@
...
  size_t Align\br
  = \textit{default-alignment}>\br
  struct aligned_storage;}
  &
- The value of \textit{default-alignment} shall be the most
+ The value of \textit{default-alignment} is the most
  stringent alignment requirement for any \Cpp{} object type whose size
  is no greater than \tcode{Len}\iref{basic.types}.
- The member typedef \tcode{type} shall be a trivial standard-layout type
+ The member typedef \tcode{type} is a trivial standard-layout type
```
Comment (zygoloid) [inline source/utilities.tex]: "denotes" here

File: source/utilities.tex [hunk 3]
```diff
@@ -17786,7 +17786,7 @@
 \indexlibrary{\idxcode{remove_cv}}%
 \tcode{template<class T>\br
  struct remove_cv;}                 &
- The member typedef \tcode{type} shall be the same as \tcode{T}
+ The member typedef \tcode{type} is the same as \tcode{T}
```
Comment (zygoloid) [inline source/utilities.tex]: Maybe "`type` denotes the type formed by removing any top-level cv-qualifiers from `T`"

File: source/utilities.tex [hunk 4]
```diff
@@ -18014,10 +18015,10 @@
   class... Types>\br
   struct aligned_union;}
   &
-  The member typedef \tcode{type} shall be a trivial standard-layout type suitable for use as
+  The member typedef \tcode{type} is a trivial standard-layout type suitable for use as
```
Comment (zygoloid) [inline source/utilities.tex]: "denotes"

**Comment by jwakely** [comment]
Rebased and a new commit added to make [refwrap.unwrapref] use "denotes the type" too.

File: source/utilities.tex [hunk 5]
```diff
@@ -17966,16 +17967,17 @@
...
+ otherwise, it denotes the type \tcode{T}.\\ \rowsep
 
 \indexlibrary{\idxcode{add_pointer}}%
 \tcode{template<class T>\br
  struct add_pointer;}                       &
- If \tcode{T} names a referenceable type\iref{defns.referenceable} or a
+ If \tcode{T} is a referenceable type\iref{defns.referenceable} or a
  \cv{}~\tcode{void} type then
- the member typedef \tcode{type} names the same type as
+ the member typedef \tcode{type} denotes the same type as
```
Comment (CaseyCarter) [inline source/utilities.tex]: "same" and "as" seem to be noise words.

File: source/utilities.tex [hunk 6]
```diff
@@ -18027,18 +18030,18 @@
 \indexlibrary{\idxcode{remove_cvref}}%
 \tcode{template<class T>\br struct remove_cvref;}
  &
- The member typedef \tcode{type} names the same type as
+ The member typedef \tcode{type} denotes the same type as
```
Comment (CaseyCarter) [inline source/utilities.tex]: Ditto "same" and "as" are noise.

File: source/utilities.tex [hunk 7]
```diff
@@ -14287,8 +14287,9 @@
 \pnum
 If \tcode{T} is
 a specialization \tcode{reference_wrapper<X>} for some type \tcode{X},
-the member typedef \tcode{type} of \tcode{unwrap_reference<T>} is \tcode{X\&},
-otherwise it is \tcode{T}.
+the member typedef \tcode{type} of \tcode{unwrap_reference<T>}
+denotes the type \tcode{X\&},
+otherwise \tcode{type} denotes the type \tcode{T}.
```
[the remark this answers is not in the document]
Comment (jwakely) [reply source/utilities.tex:14292]: The decision recorded at https://github.com/cplusplus/draft/pull/2354#issuecomment-466285910 was to say "denotes the type" but I agree that saying it's a type is redundant in many cases.

File: source/utilities.tex [hunk 8]
```diff
@@ -17915,9 +17916,9 @@
 \indexlibrary{\idxcode{remove_extent}}%
 \tcode{template<class T>\br
  struct remove_extent;}                 &
- If \tcode{T} names a type ``array of \tcode{U}'',
- the member typedef \tcode{type} shall
- be \tcode{U}, otherwise \tcode{T}.
+ If \tcode{T} is a type ``array of \tcode{U}'',
+ the member typedef \tcode{type} is \tcode{U},
```
[the remark this answers is not in the document]
Comment (jwakely) [reply source/utilities.tex:17920]: Yes, I'll fix it.

File: source/utilities.tex (hunk 5 above)
Comment (jwakely) [reply source/utilities.tex:17978]: Fixed.

File: source/utilities.tex (hunk 6 above)
Comment (jwakely) [reply source/utilities.tex:18033]: Fixed.

**Comment by jwakely** [comment]
Rebased and addressed Casey's comments.

**Review by jensmaurer — APPROVED** [review]
I like it.

**Comment by tkoeppe** [comment]
@jwakely Are all of @zygoloid comments addressed? And do you want to keep all the individual commits? If so, please go ahead and merge!

**Comment by jwakely** [comment]
> @jwakely Are all of @zygoloid comments addressed?

Actually I think https://github.com/cplusplus/draft/pull/2678#discussion_r266145972 got lost when I rebased, because that wording moved to Annex D and I didn't reapply it correctly. I'll push another commit and check if anything else got lost.

**Comment by jwakely** [comment]
I'll also apply the change requested at https://github.com/cplusplus/draft/pull/2678#discussion_r266146234 to `remove_const` and `remove_volatile`, not just to `remove_cv`.

This was a pain to rebase, because every single change moved from `utilities.tex` to either `meta.tex` or `future.tex` so had to be manually resolved.

**Comment by tkoeppe** [comment]
Oh, yes, sorry about that -- thanks a lot!

**Comment by jwakely** [comment]
OK, I think all changes are present and correct.

**Comment by jwakely** [comment]
I don't think we need to keep the individual commits, I think squash+merge makes more sense. But I can rebase it to remove the FIXUP ones if you want the individual commits.

**Comment by tkoeppe** [comment]
Yes, if a smaller number, but still more than one, makes sense, then a bit of grouping and separation is also nice.

**Comment by jwakely** [comment]
OK, cleaned up a bit and rebased. I'll leave this for a day or so for other comments and then merge.

**Review by jensmaurer — APPROVED** [review]

**Comment by tkoeppe** [comment]
I actually like the individual commits. Very tidy.

**Pull request merged**